### ISSUE TITLE
 Unreal's ASTC compression can cause RenderDoc to report a large number of asserts.

### ISSUE THREAD

**Issue by hanbim520** [baldurk/renderdoc#3432]
## Description

When using ASTC compression in the project, a large number of assert logs appear if the width and height of the texture are not multiples of the ASTC block size.
![image](https://github.com/user-attachments/assets/564c141b-9245-46ba-8fa4-26ee8fcbafc6)
![WXWorkCapture_17259587012715](https://github.com/user-attachments/assets/1fad3589-50c7-45ed-adb4-9b1375b63117)

Log:
09-10 16:55:25.998  8139  8421 I renderdoc: @66e0097d00000d40@ RDOC 008139: [16:55:25] gl_texture_funcs.cpp(3788) - Log     - width :2,height :1 ,xoffset:0,yoffset:0,format:37812
09-10 16:55:25.998  8139  8421 I renderdoc: @66e0097d00000d41@ RDOC 008139: [16:55:25] gl_texture_funcs.cpp(3791) - Error   - Assertion failed: 'width % blockSize[0] == 0' 
09-10 16:55:25.998  8139  8421 I renderdoc: @66e0097d00000d42@ RDOC 008139: [16:55:25] gl_texture_funcs.cpp(3792) - Error   - Assertion failed: 'height % blockSize[1] == 0' 

## Steps to reproduce

 test case please down apk , GitHub limits the upload size of APK files to within 25MB.
https://drive.google.com/file/d/1ziIyCY2ykVuNnbulrTTtbKFXnqcaax5W/view?usp=drive_link

## Environment

* RenderDoc version:  github source code build.（ renderdoc.exe and Android arm amr64 apk）
* Operating System: android 14
* Graphics API: opengles UnrealEngine

**Comment by hanbim520** [comment]
![1](https://github.com/user-attachments/assets/06d2e8d0-b2ad-474b-bf28-dd8529e76f65)
![121 - Copy](https://github.com/user-attachments/assets/18dc60aa-d45c-4f74-b51b-0188951036c2)

**Comment by hanbim520** [comment]
Additionally, if ASTC 6x6 is used, atlas 1024x512 will also report this issue.This issue will eventually cause the Android process to experience an ANR (Application Not Responding) and crash.

**Comment by baldurk** [comment]
For steps to reproduce, as detailed in [the documentation about filing issues](https://github.com/baldurk/renderdoc/blob/v1.x/docs/CONTRIBUTING/Filing-Issues.md) please share a specific and complete reproducible test case that I can use to investigate the problem, such as a capture or executable depending on what is most relevant to the bug. Without a repro case it is usually impossible for me to replicate the situation you are running into, and it can easily waste time trying to reproduce a problem that is not as expected or described.

Please ensure that you have run your program on its own (not from RenderDoc) with API validation and have checked that your program does not produce any validation errors or warnings, since RenderDoc does not handle invalid API use. Make sure that you have updated your graphics drivers and if you have a different GPU or hardware available try testing on that to eliminate your hardware as a potential problem.

If you can't share your test case publicly due to privacy or NDA concerns you can contact me [via email](mailto:baldurk@baldurk.org). Even if you can't share your test case privately email is typically the best avenue if you can discuss more details than you can privately.

**Comment by hanbim520** [comment]
I have uploaded the test APK to Google Drive because GitHub limits APK uploads to within 25MB, so it couldn't be uploaded here. Additionally, I noticed the submission of the log code related to RenderDoc.  
https://drive.google.com/file/d/1ziIyCY2ykVuNnbulrTTtbKFXnqcaax5W/view?usp=drive_link
![195373f08f1c97480f9bc6ba131d9088](https://github.com/user-attachments/assets/a3cda15a-1d26-4b9a-9c19-7306a0967a7b)

**Comment by hanbim520** [comment]
the issue has been confirmed to originate from RenderDoc.

**Comment by cmannett85-arm** [comment]
Hi @hanbim520, there's a mismatch between the ASTC spec and the asserts in RD.  We already have a patch to correct the asserts, but haven't had a chance to test it yet - I'll do that today.

**Comment by cmannett85-arm** [comment]
> Additionally, if ASTC 6x6 is used, atlas 1024x512 will also report this issue.This issue will eventually cause the Android process to experience an ANR (Application Not Responding) and crash.

Does the ANR occur only with RenderDoc running?  Also what is 'atlas 1024x512'?

**Comment by hanbim520** [comment]
yes ,anr occur only with renderdoc running。I have already analyzed the Android ANR logs, and the cause of the ANR is due to RenderDoc.（log from android system. /data/anr  or  /data/system/dropbox/）


'atlas 1024x512  means atlas size is 1024x512

**Comment by hanbim520** [comment]
renderdoc anr  issue  has test  on unreal 4.26 4.27 tow version.

**Comment by cmannett85-arm** [comment]
> I have already analyzed the Android ANR logs, and the cause of the ANR is due to RenderDoc.（log from android system. /data/anr or /data/system/dropbox/）

Please post this log.  I'm not seeing either the app or RD's server ANR but I am seeing a lot of GL errors on replay of the form:
```
Log     - Got a Debug message from GL_DEBUG_SOURCE_API, type GL_DEBUG_TYPE_ERROR, ID 11, severity GL_DEBUG_SEVERITY_HIGH:
Log     - 'Error:glGetIntegerv::<pname> is not an accepted value'
```
The replay app also displays blocky flickering artifacts, but the rendered output on the host is fine.

**Comment by hanbim520** [comment]
[data_app_anr@1726738735454.txt](https://github.com/user-attachments/files/17057278/data_app_anr%401726738735454.txt)
Renderdoc server is running well，but the launch app occur anr。when i close the assert log,the anr is gone.

**Comment by cmannett85-arm** [comment]
I guess I'm not seeing the ANR because I'm running your test app:
```
Activity: com.xindong.torchlight/com.epicgames.ue4.GameActivity
```

**Comment by hanbim520** [comment]
yes,I cannot provide this app as it is a commercial project of the company. I can provide the complete adb log instead.
[log.zip](https://github.com/user-attachments/files/17057555/log.zip)

**Comment by cmannett85-arm** [comment]
Your log is very confused, I've pulled out some key moments:
```
09-19 17:15:56.573  2690  2736 I ActivityManager: Start proc 11533:org.renderdoc.renderdoccmd.arm64/u0a394 for next-top-activity {org.renderdoc.renderdoccmd.arm64/org.renderdoc.renderdoccmd.arm64.Loader} caller=com.android.shell

09-18 17:17:04.878  2690  2736 I ActivityManager: Start proc 22800:com.xindong.torchlight/u0a388 for next-top-activity {com.xindong.torchlight/com.epicgames.ue4.SplashActivity} caller=com.miui.home
09-18 17:17:06.596  2690  8123 I ActivityTaskManager: The Process com.xindong.torchlight Already Exists in BG. So sending its PID: 22800

09-18 17:17:10.610  2690  2736 I ActivityManager: Start proc 23093:com.google.android.webview:sandboxed_process0:org.chromium.content.app.SandboxedProcessService0:0/u0i7 for  {com.xindong.torchlight/org.chromium.content.app.SandboxedProcessService0:0} caller=com.xindong.torchlight

09-19 17:17:35.812  2690  2736 I ActivityManager: Start proc 13134:com.xindong.torchlight/u0a388 for next-top-activity {com.xindong.torchlight/com.epicgames.ue4.SplashActivity} caller=com.android.shell
09-19 17:17:36.069  2690  8124 I ActivityTaskManager: The Process com.xindong.torchlight Already Exists in BG. So sending its PID: 13134

09-19 17:17:39.399  2690  2736 I ActivityManager: Start proc 13422:com.google.android.webview:sandboxed_process0:org.chromium.content.app.SandboxedProcessService0:0/u0i9 for  {com.xindong.torchlight/org.chromium.content.app.SandboxedProcessService0:0} caller=com.xindong.torchlight

09-19 17:17:42.471 13134 13159 W MIUIScout App: Enter APP_SCOUT_WARNING State
09-19 17:17:42.472 13134 13159 W MIUIScout App:  (Current message: duration=2503ms seq=74 h=android.view.Choreographer$FrameHandler c=android.view.Choreographer$FrameDisplayEventReceiver)
09-19 17:17:42.474 13134 13159 D AppScoutStateMachine: dump trace dirName = 2024-09-19-17-17-42-13134-APP_SCOUT_WARNING-com.xindong.torchlight
09-19 17:17:42.476 13134 13521 W MIUIScout App: Event:APP_SCOUT_WARNING Thread:main backtrace:
09-19 17:17:42.476 13134 13521 W MIUIScout App:     at android.graphics.HardwareRenderer.nSyncAndDrawFrame(Native Method)

09-19 17:17:44.999  2690  4064 I WindowManager: ANR in Window{b098012 u0 com.xindong.torchlight/com.epicgames.ue4.GameActivity}. Reason:Input dispatching timed out (b098012 com.xindong.torchlight/com.epicgames.ue4.GameActivity (server) is not responding. Waited 5000ms for FocusEvent(hasFocus=true))

09-18 17:25:30.278  2690  8035 I ActivityManager: Process com.xindong.torchlight (pid 22800) has died: fg  TOP 

```
So it looks like you:

1. Started RD
2. Launched your app from the home launcher
3. Launched it again via RD - I don't know if this is a problem but it certainly isn't helping
 
Please kill your app and RD, then clear logcat, and then the run the test _once_ and attach the logcat output to this issue.  I also noticed that your app spawns a webview component, I'd be curious to see what happens if you disable this behaviour - I don't think it will affect the ANR but it's worth a test.

**Comment by hanbim520** [comment]
I tried disabling the WebView, and the ANR no longer occurs. I suspect that this WebView SDK may have some conflict with RenderDoc. I'll put the ANR issue aside for now and check if there is a problem with the SDK. This SDK is a third-party one.

**Comment by hanbim520** [comment]
Additionally, disabling the assert logs in RenderDoc also prevents the ANR from occurring. I suspect that there might be a race condition related to the logging part that is causing this issue.

**Comment by cmannett85-arm** [comment]
That's good to hear, at least you have a work around.  Once the associated PR for the asserts is merged I'll close this issue.  The GL errors I'm seeing I'll investigate as a separate issue.

Thanks!

**Comment by hanbim520** [comment]
Likewise, thank you for your technical support!
